sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +742,62 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, p):
+    """
+    Find the solutions to ``x**n = a mod p`` when p is not prime.
+    """
+    f = factorint(p)
+    dd = {}
+    for b, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            for num in nthroot_mod(a, n, b, True):
+                tot_roots.add(num)
+                dd[b] = tot_roots
+            continue
+        else:
+            roots = nthroot_mod(a, n, b, True)
+            if roots == None:
```

Comment:
`roots is None` will be more efficient.

Reply:
(as above, these two lines can be deleted if the modification is made)